### DISCUSSION TITLE
"is_action_pressed" should be "is_action_held"

### DISCUSSION THREAD

**Discussion by Jago1996** [godotengine/godot-proposals#7585] [Scripting]
Makes the code more readable. Had a few people talking about it in discord and it really would just make more sense. I'm unsure why it wasn't this from the get-go. 
  
I can understand how some names would be too unwieldy if we didn't keep them short and just descriptive enough to understand. But in this case is_action_held would just make complete sense and it even saves you some letters.

**Comment by timothyqiu** [comment]
A button is either pressed or not pressed. But if "hold" is introduced, "just pressed" is usually not "held". So to get the current "is_action_pressed", you would need to check both pressed and held.

**Reply by Jago1996** [reply]
I may be misunderstanding then. The difference between is_action_pressed and is_action_just_pressed is that is_action_pressed fires if that action key is held down, where as is_action_just_pressed fires only on the first instance that action is pressed.  
  
 Would my reasoning not make sense? is_action_just_pressed would still be named the same, where as is_action_held would just be the new way to do is_action_pressed, I think it makes more sense this way.

**Reply by timothyqiu** [reply]
The difference you described is correct.

The current naming makes sense because only "pressed" is used, so you know it's referring to a state. `is_action_pressed` includes `is_action_just_pressed`.

But when "press" and "hold" both appears, people usually prefer them to be separate actions, i.e., pressed = the button is pushed down on this frame, and held = the button is pushed down on previous frames and it is still pressed. Hold does not include press. So the old `is_action_pressed` would become `is_action_just_pressed() and is_action_held()`.

**Comment by theraot** [comment]
I would expect `Input.is_action_held("action")` to mean `Input.is_action_pressed("action") and not Input.is_action_just_pressed("action")`. That is: it is not just pressed, it has been pressed for at least one frame (to reiterate: it is pressed and not just pressed this frame).
